### DISCUSSION TITLE
Column Alias

### DISCUSSION THREAD

**Discussion by yeanpei** [simolus3/drift#2479] [Q&A] [answered]
I want to select all columns in primary table (`issue`), and only certain columns in the join tables. 
In SQL would be `SELECT issue.*, cat.desc, blocks.desc, assignee.desc, creator.desc`

1. How do I write for `issue.*` inside addColumns when using `selectOnly`? Or do I need to specify all columns like `issue.a, issue.b...` ? 
```
 final query = selectOnly(issue).join([
      innerJoin(cat, issue.categoryKey.equalsExp(cat.key), useColumns: false),
      innerJoin(blocks, issue.blockKey.equalsExp(blocks.key), useColumns: false),
      innerJoin(assignee, issue.assigneeKey.equalsExp(assignee.key), useColumns: false),
      innerJoin(creator, issue.createdBy.equalsExp(creator.key), useColumns: false)
    ])
      ..where(issue.uuid.equalsExp(subqueryExpression(groupId)))
      ..addColumns([cat.desc, blocks.desc, assignee.desc, creator.desc]);
```

2. How to write for column name alias? 
In SQL, we can write like `cat.desc AS CatDesc`.
I check the log SQL it auto alias as `cat.desc AS 'cat.desc'`, but I want to rename myself, is it possible?

**Comment by simolus3** [comment, marked as answer]
1. I would use the snippet you've posted, yeah. That selects every column from `issue` and the additional columns from the joined tables.
> 2. but I want to rename myself, is it possible?

No. Do you have a use case where it would be helpful to assign a custom name to that column? Drift manages the name for you so that you can use `row.read(cat.desc)` in the results. So the alias managed by drift should be seen as an implementation detail and you shouldn't access the low-level row map directly for drift-generated queries. I wouldn't consider it breaking if drift suddenly changed the alias as well.
(If you have a compelling scenario where that would be useful, I can look into ways we could allow explicit aliases)

**Reply by yeanpei** [reply]
I thought of passing the result as **json** from **rawData** to another view model class to deserialize using the json key specified (where I want to rename column to be same as json key). I realized I could do this `row.read(cat.desc)` just that I need to assign column by column.
